src/store.cc

```diff
@@ -755,14 +760,14 @@ StoreEntry::write (StoreIOBuffer writeBuffer)
 {
     assert(mem_obj != nullptr);
     /* This assert will change when we teach the store to update */
-    assert(store_status == STORE_PENDING);
```

Comment:
Can we keep this assertion? It feels like it was correct or, at the very least, should not be removed in _this_ PR.